packages/core/utils/lib/async.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+function pipeAsync(...methods) {
+  return async (data) => {
+    let res = data;
+
+    for (const method of methods) {
+      res = await method(res);
+    }
+
+    return res;
+  };
+}
+
+/**
+ * Map function callback.
+ * @callback mapAsyncCallback
+ * @param {*} value
+ * @param {number} index
+ */
+/**
+ * Async iteration over an array of promises
+ * @param {promise<*>[]} promiseArray
+ * @returns {function(callback: mapAsyncCallback): promise<*[]>}
```

Comment:
The function only returns one promises not an array of promises

Reply:
Good catch, thanks

Reply:
Actually, it says that it returns a single promise with an array of any inside it. 🤔

Reply:
Do you want me to do a typescript declaration file to be a bit more clear on typings ? @alexandrebodin

Reply:
Yes that's `Promise<*[]>` not `Promise<*>[]`

Reply:
Yes if you are up for it you can do a declaration file and import the typings in the js 👍

Reply:
Should be good now 👍